### ISSUE TITLE
Working with EOL (End Of Line)

### ISSUE THREAD

**Issue by username1565** [WinMerge/winmerge#643]
Hello.
I have three files, the same code, but with different EOL.
"CR" (Mac), "LF" (Unix), and "CRLF" (Windows).
When I try to compare this, I see differences in all the lines, but code is the same.

When files are different, and just one line is different for each file,
I can not select THIS difference, as one difference, just because different all lines, and this is as one difference.

I see just one hint, about EOL, (Mac, Win, Unix), on the screen.
![image](https://user-images.githubusercontent.com/39200328/108788229-64eceb00-756f-11eb-8dd0-307b92c589b5.png)

After open files, in `Notepad++`, and after press the button 
![image](https://user-images.githubusercontent.com/39200328/108788293-8cdc4e80-756f-11eb-9441-0e9559947c66.png)
I can see EOL, there. But this so hard when there is many files to compare,
on try to compare folder, for example.

Can you add the ability to make the following things:
1. Display and show the EOL, like in `NotePad++` (optionally, and disabled by default, but allow to enable/disable).
2. Optionally ignore EOL-differences, and allow to work with line-differences (move this between files),
  and work with this line-differences, as with separate differences, not as with one difference in all lines with different EOL.
  Maybe this already implemented, but I do no know how to use this.
3. Add ability to change EOL for all lines just by one click, for example there, where (Mac, Win, Unix is displayed).
    There can be opened select-menu, with checkbox, by one click, to allow to change EOL for all lines in specified file.
4. Add ability to change EOL for group of files, in folder and subfolders, without open this files (this can eat many memory just for opening this files).

**Comment by gerundt** [comment]
For **point 2** try under `Options -> Compare -> General` the setting `Ignore carriage return differences`:

![image](https://user-images.githubusercontent.com/497880/108982869-8d1f3b80-768e-11eb-8e99-61b1411428cb.png)

**Comment by sdottaka** [comment]
For point 1, check "Keep original EOL characters" in the "Editor" category of the Options dialog, and check the "View"-> "Show EOL" menu item. By doing so, you can distinguish whether the EOL is CRLF or LF.

![image](https://user-images.githubusercontent.com/98126/109374019-bee60b80-78f5-11eb-8163-8cbe5ca17843.png)

![image](https://user-images.githubusercontent.com/98126/109374056-0e2c3c00-78f6-11eb-89e9-45202ef032b0.png)

For point 3, it's not a single click, but you can double-click Unix and click Windows to make the EOL CRLF, as shown below.

![image](https://user-images.githubusercontent.com/98126/109374640-1c7c5700-78fa-11eb-87a8-a3181c33619f.png)

**Comment by username1565** [comment]
@gerundt
> For **point 2** try under `Options -> Compare -> General` the setting `Ignore carriage return differences`

Nice, thanks. I found it there: 
![image](https://user-images.githubusercontent.com/39200328/109644739-cacc0a80-7b4d-11eb-8926-178df5a98b61.png)

@sdottaka 
> For **point 1**, check `"Keep original EOL characters"` in the `"Editor"` category of the `Options dialog`, and check the `"View"-> "Show EOL"` menu item. By doing so, you can distinguish whether the `EOL` is `CRLF` or `LF`.

Thanks! In v2.16.7, I found this, just as `View WhiteSpaces` option:
![image](https://user-images.githubusercontent.com/39200328/109645946-732e9e80-7b4f-11eb-9406-a245613a7b84.png)

> For **point 3**, it's `not a single click`, but you can `double-click` Unix and click Windows to make the `EOL CRLF`, as shown below.

Thanks. I not found this in v2.16.7, but in v2.16.10 this is already working. I've already updated my Winmerge up to v2.16.10.

> 4. Add ability to change EOL for group of files, in folder and subfolders,
>without open this files (this can eat many memory just for opening this files).

It's not necessarry to implement, because I found the better way, using `Notepad++`.
1. Just drag and dtop folder with files into `new Instance of Notepad++`.
2. Many files opened in one window.
3. Click π -button to see EOL.
4. Then, for each file - 3 clicks:
![image](https://user-images.githubusercontent.com/39200328/109646919-9ad23680-7b50-11eb-822f-f5616dc3957c.png)

I close this issue, because all points was been resolved.
Thanks everyone! Great program.
